### PR TITLE
Fix/format time indicator

### PR DESCRIPTION
**Pull request by frasercl**
Review time: smallish (~10-15min) (large diff numbers are package-lock.json!)

Addresses allen-cell-animated/website-3d-cell-viewer#202: adapts some of @ShrimpCryptid's code from nucmorph-colorizer to format numbers more nicely and keep them from spilling into each other too much if they have very long decimals.

Also: resolves issues I was having with `npm link`ing to volume-viewer to verify the above worked by disabling and removing most of Babel's major build transformations. The various polyfills it was giving us are more appropriate for a published site anyways, rather than a library which is likely to be transformed by another downstream build process.

### To test:

- `npm run build`
- `npm link` to website-3d-cell-viewer
- `npm start` website-3d-cell-viewer
- open http://localhost:9020/?url=https://dev-aics-dtp-001.int.allencell.org//assay-dev/computational/data/release/nuc-morph-data/data_for_analysis/baseline_colonies/20200323_09_small/raw.ome.zarr
- switch to XY mode

**On `main`**: time indicator and scale bar overlap as in the screenshot in allen-cell-animated/website-3d-cell-viewer#202

**On this branch**: time indicator is well formatted:
image

### CODE REVIEW

File: src/utils/num_utils.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+function trimTrailing(str: string, char: string): string {
+  let i = str.length - 1;
+  while (str[i] === char) {
+    i--;
+  }
+  return str.slice(0, i + 1);
+}
+
+export function formatNumber(value: number): string {
+  if (value < 0.01 || value >= 10_000) {
```
Comment (ShrimpCryptid) [inline src/utils/num_utils.ts:46]: Should this also do a check for values in the range `[-0.01, 0]` and `<= -10_000` ?

(I think this function isn't expected to deal with negative numbers in the current use case but might be nice to be ready for!)

File: src/utils/num_utils.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return numberToSciNotation(value, 3);
+  } else if (Number.isInteger(value)) {
+    return value.toString();
+  } else {
+    // `toPrecision` may try to format numbers in scientific notation, so we do a similar thing with `toFixed` instead.
+    const numStr = value.toFixed(DEFAULT_SIG_FIGS - Math.floor(Math.log10(value)) - 1);
+    const trimmed = trimTrailing(numStr, "0");
+    return trimmed.endsWith(".") ? trimmed.slice(0, -1) : trimmed;
+  }
+}
```
Comment (ShrimpCryptid) [inline src/utils/num_utils.ts:60]: Might be a good idea to have some unit tests for this to make sure numbers are formatted how you expect them to be? I have some unit tests here for `numberToSciNotation` if you want to use them as a basis.

https://github.com/allen-cell-animated/nucmorph-colorizer/blob/main/tests/math_utils.test.ts#L5-L46

File: src/ThreeJsPanel.ts [hunk 3]
```diff
@@ -416,22 +417,23 @@ export class ThreeJsPanel {
...
-    let scaleStr = scaleValue.toString();
-    if (digits < 1) {
-      // Handle irrational floating point values (e.g. 0.30000000000000004)
-      scaleStr = scaleStr.slice(0, Math.abs(digits) + 2);
-    }
+    // let scaleStr = scaleValue.toString();
+    // if (digits < 1) {
+    //   // Handle irrational floating point values (e.g. 0.30000000000000004)
+    //   scaleStr = scaleStr.slice(0, Math.abs(digits) + 2);
+    // }
```
[the remark this answers is not in the document]
Comment (frasercl) [reply src/ThreeJsPanel.ts:424]: yup! I think the commit to do this got lost in shuffling this branch between my laptop and desktop.

File: src/utils/num_utils.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const expSign = exponent[0] === "-" ? "⁻" : "";
+  const expDigits = exponent.slice(1).split("");
+  const expSuperscript = expDigits.map((digit) => digitToUnicodeSupercript(Number(digit))).join("");
+  return `${significand}×10${expSign}${expSuperscript}`;
+}
+
+const ZERO_SCI = numberToSciNotation(0);
+
+/** Trims trailing instances of `char` off the end of `str`. */
+// This is not technically a number utility, but it's useful to `formatNumber` below.
```
Comment (toloudis) [inline src/utils/num_utils.ts:36]: Awesome combination of commenting styles.

File: src/utils/num_utils.ts (hunk 1 above)
Comment (toloudis) [reply src/utils/num_utils.ts:46]: It does seem like this should maybe do the check for abs(value) or >0 if you can't handle negatives.

**Review by toloudis — APPROVED** [review]
Awesome changes.  I agree with Peyton that maybe some edge cases might want a unit test to confirm behavior but it's not a showstopper for this, imo.

File: src/utils/num_utils.ts (hunk 1 above)
Comment (frasercl) [reply src/utils/num_utils.ts]: Good catch! Added.

File: src/utils/num_utils.ts (hunk 2 above)
Comment (frasercl) [reply src/utils/num_utils.ts]: Added some unit tests, then tightened the function up in response to the unit tests, then added more unit tests for good measure. Check it out!

File: src/test/num_utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    });
+
+    it("formats numbers which would round to 10000 in scientific notation", () => {
+      expect(formatNumber(9999.9)).to.equal("9999.9");
+      expect(formatNumber(9999.99)).to.equal("1.00×10⁴");
+    });
+
+    it("does not account for numbers which round up to 0.01 in scientific notation", () => {
+      expect(formatNumber(0.00999)).to.equal("9.99×10⁻³");
+      expect(formatNumber(0.009999)).to.equal("1.00×10⁻²"); // aw shucks
```
Comment (ShrimpCryptid) [inline src/test/num_utils.test.ts:46]: Maybe add a TODO on these? But they're also minimal corner cases so I totally get skipping them.

**Review by ShrimpCryptid — APPROVED** [review]
Looks good, love the explicit documentation and the new unit tests!
